server/src/main/java/org/elasticsearch/index/analysis/Analysis.java

```diff
@@ -267,12 +267,18 @@ public static List<String> getWordList(
         Settings settings,
         String settingPath,
         String settingList,
+        String settingDeduplicate,
         boolean removeComments,
         boolean checkDuplicate
     ) {
+        boolean deduplicateDictionary = settings.getAsBoolean(settingDeduplicate, false);
         final List<String> ruleList = getWordList(env, settings, settingPath, settingList, removeComments);
         if (ruleList != null && ruleList.isEmpty() == false && checkDuplicate) {
-            checkDuplicateRules(ruleList);
+            if (deduplicateDictionary) {
+                return ruleList.stream().distinct().toList();
```

Comment:
I don't think this will work. The deduplication logic should be the exact same as the `checkDuplicateRules` logic. 

For example:

```
                String[] values = CSVUtil.parse(line);

```
And then only the first value is checked for duplication. Your code here could have `foo,bar` and `foo,baz` pass through distinct stream. That doesn't seem correct.

Reply:
good feedback.  I agree let me unify the checkDuplicateRules function so they behave the same and to alleviate any of your other concerns.